### PR TITLE
Added basic SecurityConfig

### PR DESCRIPTION
**Pull request by rasmusbaeckstrom**
Basic Security implemented. 

Now all requests are allowed (permitAll) and endpoints can be added as the project progresses.

OAuth2 GitHub login will replace httpBasic.

### CODE REVIEW

**Review by sebastianpiresmolin — DISMISSED** [review]
Awesome Rasmus, now we can start setting up pathing  and access rules.

Just make the bot happy and go!

**Comment by rasmusbaeckstrom** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=fungover_system2024&pullRequest=45) **Quality Gate failed**
> Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=fungover_system2024&pullRequest=45&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)
> 
> [See analysis details on SonarQube Cloud](https://sonarcloud.io/dashboard?id=fungover_system2024&pullRequest=45)

@kappsegla 
This is a new area for me. What can I do here? or do you, who have admin rights, need to do something?
From what I understand, it should be okay to have CSRF disabled during the development stage.

**Review by MickeMarten — APPROVED** [review]
Great initiative. Good ground to build from.

**Comment by kappsegla** [comment]
> > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=fungover_system2024&pullRequest=45) **Quality Gate failed**
> > Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=fungover_system2024&pullRequest=45&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)
> > [See analysis details on SonarQube Cloud](https://sonarcloud.io/dashboard?id=fungover_system2024&pullRequest=45)
> 
> @kappsegla This is a new area for me. What can I do here? or do you, who have admin rights, need to do something? From what I understand, it should be okay to have CSRF disabled during the development stage.

Do we really need to disable csrf? Our client will be our own frontend and if we are going to use javascript for accessing the api to get data we can use csrf by getting a csrf token from a protected endpoint that only our code can access and then other scripts can't issue http requests that will alter data on the server.

Using csrf protection with single page applications is talked about a bit here:
https://docs.spring.io/spring-security/reference/servlet/exploits/csrf.html#csrf-integration-javascript-spa

**Comment by oskber** [comment]
Looks good, maybe enable csrf as @kappsegla is suggesting?
My GitHub Oauth2 login implementation is ready for a PR, but I will wait for this PR to merge and then refactor/fix conflicts I know will appear before I push.

File: src/main/java/org/fungover/system2024/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Bean
+    @Profile("development")
+    public SecurityFilterChain developmentSecurityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf(Customizer.withDefaults()) // Enable CSRF protection
```
Comment (sebastianpiresmolin) [inline src/main/java/org/fungover/system2024/config/SecurityConfig.java:19]: I see that you enables csrf instead. Like martin said, It's going to be used evenutally anyway. Good job!

**Review by sebastianpiresmolin — APPROVED** [review]

File: src/main/java/org/fungover/system2024/config/SecurityConfig.java (hunk 1 above)
Comment (kappsegla) [reply src/main/java/org/fungover/system2024/config/SecurityConfig.java:19]: And it won't break anything for now. If we get into problems in the future we deal with it then...

**Review by MickeMarten — APPROVED** [review]

**Pull request merged**